DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/EncodedBlocks.h

```diff
@@ -125,15 +130,42 @@ struct Block {
 
   Registry* registry = nullptr; //! non-persistent info for in-memory ops
   int nDict = 0;                // dictionary length (if any)
-  int nStored = 0;              // total payload: data + dictionary length
+  int nData = 0;                // length of data
+  int nLiterals = 0;            // length of literals vector (if any)
+  int nStored = 0;              // total length
   W* payload = nullptr;         //[nStored];
 
-  W* getData() { return payload ? (payload + nDict) : (payload = reinterpret_cast<W*>(registry->getFreeBlockStart())); }
-  W* getDict() { return nDict ? payload : nullptr; }
-  const W* getData() const { return payload ? (payload + nDict) : nullptr; }
   const W* getDict() const { return nDict ? payload : nullptr; }
```

Comment:
@shahor02 I don't like that overload sets for getData for const and non-const do different things and will error differently. Is there a way to make the registry set the pointer correctly on initialization?

Reply:
The pointer is not fixed, it moves together with block filling. I don't see problem with const/non-const doing different things, but if you, you can rename the non-const to e.g. getCreateDataPointer...

Reply:
Fixed after discussion